packages/expo-contacts/android/src/main/java/expo/modules/contacts/Contact.kt

```diff
@@ -285,15 +285,31 @@ class Contact(var contactId: String) {
           .build()
       )
     }
+
+    // Flush all data from linked db
+    ops.add(getFlushOperation(CommonDataKinds.Event.CONTENT_ITEM_TYPE, rawContactId!!))
+    ops.add(getFlushOperation(CommonDataKinds.Email.CONTENT_ITEM_TYPE, rawContactId!!))
+    ops.add(getFlushOperation(CommonDataKinds.Im.CONTENT_ITEM_TYPE, rawContactId!!))
+    ops.add(getFlushOperation(CommonDataKinds.Phone.CONTENT_ITEM_TYPE, rawContactId!!))
+    ops.add(getFlushOperation(CommonDataKinds.StructuredPostal.CONTENT_ITEM_TYPE, rawContactId!!))
+    ops.add(getFlushOperation(CommonDataKinds.Relation.CONTENT_ITEM_TYPE, rawContactId!!))
+    ops.add(getFlushOperation(CommonDataKinds.Website.CONTENT_ITEM_TYPE, rawContactId!!))
+    ops.add(getFlushOperation(CommonDataKinds.Nickname.CONTENT_ITEM_TYPE, rawContactId!!))
+    // add updated data
     for (map in baseModels) {
       for (item in map) {
-        ops.add(item.getDeleteOperation(rawContactId!!))
         ops.add(item.getInsertOperation(rawContactId))
       }
     }
     return ops
   }
 
+  private fun getFlushOperation(contentType: String, rawId: String): ContentProviderOperation? {
+      return ContentProviderOperation.newDelete(ContactsContract.Data.CONTENT_URI)
+        .withSelection(String.format("%s=? AND %s=?", ContactsContract.Data.MIMETYPE, ContactsContract.Data.RAW_CONTACT_ID), arrayOf(contentType, rawId))
+        .build()
+  }
```

Comment:
```suggestion
private fun getFlushOperation(contentType: String, rawId: String): ContentProviderOperation {
  return ContentProviderOperation.newDelete(ContactsContract.Data.CONTENT_URI)
    .withSelection("${ContactsContract.Data.MIMETYPE}=? AND ${ContactsContract.Data.RAW_CONTACT_ID}=?", arrayOf(contentType, rawId))
    .build()
}
```
We don't need String.format here and the return type won't need to be nullable

Reply:
Ok, I integrate this fix as is. Thank you